pkg/v1/match/match_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package match_test
```

Comment:
I don't mind these being in a separate _test package, but we tend to do that only when the tests would otherwise introduce a dependency cycle, so I have a (very) weak preference for keeping these in `match` only for the sake of consistency.

Reply:
Funny you say that. I have a preference for separate testing packages, as it forces testing the API surface as consumers would see it, rather than the internals. If I need also to test internals, then I do so in `match_internal_test.go`, which would have `package match`.

But I am hardly dogmatic about it. I can move it over if you want.

Reply:
I'm okay with keeping as is :)